toolchain/parse/handle_var.cpp

```diff
@@ -6,22 +6,41 @@
 
 namespace Carbon::Parse {
 
-// Handles VarAs(Semicolon|For).
+// Handles VarAs(Declaration|For).
 static auto HandleVar(Context& context, State finish_state) -> void {
   context.PopAndDiscardState();
 
-  // These will start at the `var`.
+  // These will start at the `var` or `returned`.
   context.PushState(finish_state);
   context.PushState(State::VarAfterPattern);
 
+  // Consume the `var` or `returned var`.
+  auto returned_token = context.ConsumeIf(Lex::TokenKind::Returned);
```

Comment:
Given the `Returned` is already seen from the caller, had you considered a separate state for it so that this wouldn't need to check the token kind again?

Reply:
I did try that, but previously had been trying to keep this `ConsumeIf` after the `PushState`s, which made it awkward. Seems to work a bit better now. Done.